From 68c9802ee13c4d3ef16cb69a661ebf6b567a502b Mon Sep 17 00:00:00 2001
From: Zhi Wang <zhi.a.wang@intel.com>
Date: Tue, 21 Apr 2015 08:28:07 +0800
Subject: [PATCH 046/403] i915: move pending bdw WAs into proper place.

Do know why patch command apply these pieces into another functions.

Signed-off-by: Zhi Wang <zhi.a.wang@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |   18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 1a23713..8fe165d 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -6513,6 +6513,15 @@ static void broadwell_init_clock_gating(struct drm_device *dev)
 	 */
 	I915_WRITE(HSW_GTT_CACHE_EN, GTT_CACHE_EN_ALL);
 
+	/* WaOCLCoherentLineFlush:bdw */
+	I915_WRITE(GEN8_L3SQCREG4, I915_READ(GEN8_L3SQCREG4) |
+		   GEN8_PIPELINE_FLUSH_COHERENT_LINES);
+
+	/* WaDisableMidThreadPreempt:bdw */
+	I915_WRITE(GEN8_FF_SLICE_CS_CHICKEN2,
+		   I915_READ(GEN8_FF_SLICE_CS_CHICKEN2) |
+		   _MASKED_BIT_ENABLE(GEN8_THREAD_GROUP_PREEMPTION));
+
 	I915_WRITE(0xb10c, (I915_READ(0xb10c) & ~(0xf << 20)) | (0x8 << 20));
 
 	lpt_init_clock_gating(dev);
@@ -6571,15 +6580,6 @@ static void haswell_init_clock_gating(struct drm_device *dev)
 	I915_WRITE(CHICKEN_PAR1_1,
 		   I915_READ(CHICKEN_PAR1_1) | FORCE_ARB_IDLE_PLANES);
 
-	/* WaOCLCoherentLineFlush:bdw */
-	I915_WRITE(GEN8_L3SQCREG4, I915_READ(GEN8_L3SQCREG4) |
-		   GEN8_PIPELINE_FLUSH_COHERENT_LINES);
-
-	/* WaDisableMidThreadPreempt:bdw */
-	I915_WRITE(GEN8_FF_SLICE_CS_CHICKEN2,
-		   I915_READ(GEN8_FF_SLICE_CS_CHICKEN2) |
-		   _MASKED_BIT_ENABLE(GEN8_THREAD_GROUP_PREEMPTION));
-
 	lpt_init_clock_gating(dev);
 }
 
-- 
1.7.10.4

